File: Array & String/Insert Delete GetRandom O(1)/GetRandom.cs
public class RandomizedSet
{
    private Dictionary<int, int> indices;
    private List<int> arr;
    private Random rand;

    public RandomizedSet()
    {
        indices = new Dictionary<int, int>();
        arr = new List<int>();
        rand = new Random();
    }

    public bool Insert(int val)
    {
        if (indices.ContainsKey(val))
            return false;

        arr.Add(val);
        indices[val] = arr.Count - 1;
        return true;
    }

    public bool Remove(int val)
    {
        if (!indices.ContainsKey(val))
            return false;

        int i = indices[val];
        indices[arr[arr.Count - 1]] = i;
        arr[i] = arr[arr.Count - 1];
        indices.Remove(val);
        arr.RemoveAt(arr.Count - 1);
        return true;
    }

    public int GetRandom()
    {
        return arr[rand.Next(arr.Count)];
    }
}

File: Array & String/Insert Delete GetRandom O(1)/GetRandom.cpp
class RandomizedSet {
private:
    unordered_map<int, int> indices;
    vector<int> arr;

public:
    RandomizedSet() {
        srand(time(0));
    }

    bool insert(int val) {
        if (indices.find(val) != indices.end())
            return false;

        arr.push_back(val);
        indices[val] = arr.size() - 1;
        return true;
    }

    bool remove(int val) {
        if (indices.find(val) == indices.end())
            return false;

        int i = indices[val];
        indices[arr.back()] = i;
        arr[i] = arr.back();
        indices.erase(val);
        arr.pop_back();
        return true;
    }

    int getRandom() {
        int randIndex = rand() % arr.size();
        return arr[randIndex];
    }
};